bittide-instances/src/Bittide/Instances/Pnr/Ethernet.hs

```diff
@@ -0,0 +1,192 @@
+
+  peConfig =
+    PeConfig (0b100 :> 0b001 :> 0b010 :> 0b011 :> 0b101 :> 0b110 :> 0b111 :> 0b000 :> Nil)
+    (Undefined @(256 * 1024))
+    (Undefined @(64 * 1024))
+
+vexRiscEthernet ::
+  Clock Basic125B ->
+  Reset Basic125B ->
+  DiffClock Basic625 ->
+    ( Signal Basic125B JtagIn
+    , Signal Basic125B Bit
+    , Signal Basic625 Lvds
+    ) ->
+    ( Signal Basic125B JtagOut
+    , Signal Basic125B Bit
+    , Signal Basic625 Lvds
+    , Signal Basic125B (BitVector 32)
+    )
+vexRiscEthernet sysClk sysRst sgmiiPhyClk (jtagin, uartIn, sgmiiIn) =
+  (jtagOut, uartOut, bridgeLvdsOut, gpioOut)
+ where
+  BridgeOutput{..} = bridge sgmiiIn gmiiOut
+  signalDetect = pure True
+  anRestart = pure False
+  conf = pure def{cAutoNegEnable = True}
+  anConf = pure def{cAcknowledge=True, cDuplexMode = FullDuplex, cLinkSpeed = Speed1000, cPhyLinkStatus = True}
+  bridge = gmiiSgmiiBridge sgmiiPhyClk bridgeRst signalDetect conf anConf anRestart
+  rxClk = bridgeClk125 :: Clock Basic125A
+  rxRst = bridgeRst125
+  bridgeRst = unsafeResetDesynchronizer sysClk sysRst
+  (uartOut, gmiiOut, jtagOut, gpioOut) = vexRiscGmii SNat sysClk sysRst rxClk rxRst rxClk rxRst (uartIn, bridgeGmiiRx, jtagin)
+
+vexRiscEthernetTop ::
+  "CLK_300MHZ" ::: DiffClock Ext300 ->
+  "CPU_RESET" ::: Reset Ext300 ->
+  "sgmii_phyclk" ::: DiffClock Basic625 ->
+    ( "JTAG" ::: Signal Basic125B JtagIn
+    , "USB_UART_TXD" ::: Signal Basic125B Bit
+    , "sgmii_rx" ::: Signal Basic625 Lvds
+    ) ->
+    ( "JTAG" ::: Signal Basic125B JtagOut
+    , "USB_UART_RXD" ::: Signal Basic125B Bit
+    , "sgmii_tx" ::: Signal Basic625 Lvds
+    )
+vexRiscEthernetTop diffClk cpuReset sgmiiClk inp = (j, u , s)
+ where
+  (sysClk, sysRst) = clockWizardDifferential diffClk cpuReset
+  (j, u, s, _) = vexRiscEthernet sysClk sysRst sgmiiClk inp
+
+{-# ANN vexRiscEthernetTop (Synthesize
+  { t_name = "vexRiscEthernet"
+  , t_inputs =  [ PortProduct "CLK_125MHZ"
+                  [PortName "P", PortName "N"]
+                , PortName "CPU_RESET"
+                , PortProduct "SGMIICLK"
+                  [PortName "P", PortName "N"]
+                , PortProduct ""
+                  [ PortProduct "JTAG"
+                    [ PortName "TCK", PortName "TMS", PortName "TDI"]
+                  , PortName "USB_UART_TX"
+                  , PortProduct "SGMII"
+                    [ PortName "RX_P", PortName "RX_N"]
+                  ]
+                ]
+  , t_output =
+    PortProduct ""
+      [ PortProduct "JTAG"
+        [ PortName "TDO"
+        , PortName "RST"
+        ]
+      , PortName "USB_UART_RX"
+      , PortProduct "SGMII"
+        [PortName "TX_P", PortName "TX_N"]
+      ]
+  }) #-}
+
+
+-- TODO: Add more constraints?
+-- | Convert a reset signal from one domain to another domain and insert a register.
+-- This can be used to move a reset signal from a domain with a synchronous reset to
+-- a domain with an asynchronous reset.
+unsafeResetDesynchronizer ::
```

Comment:
It's still synchronizing, just to other specifications. I also think it's not quite correct; you want to sample in the target domain with multiple ffs _or_ specify that users should be very careful and make sure that these domains have the same clock. Unless there is a good reason to care about that single flipflop, just use `xpmResetSynchronizer`.

Reply:
Actually we don't want any logic or state in the "target" domain.
We use this for the asynchronous reset port on of `gmiiSgmiiBridge` which, according to https://docs.amd.com/v/u/en-US/gig_eth_pcs_pma_ds264 : 
| Signal | Direction | Clock Domain | Description                                          |
|--------|-----------|--------------|------------------------------------------------------|
| reset  | Input     | n/a          | Asynchronous reset for the entire core. Active High. |

The signal is not supposed to be synchronous, but we can't express that in Clash....

Reply:
`delayReset` (filter glitches) + `unsafeResetSynchronizer` (change domain)

Reply:
`unsafeResetSynchronizer = unsafeFromBla . unsafeSynchronizer . unsafeToBla`